client-side/src/components/profileComponents/profilePageComponent.jsx

```diff
@@ -0,0 +1,231 @@
+      await updateProfileApi(selectedProfile._id, profileToUpdate);
+      dispatch(updateProfile(profileToUpdate));
+      console.log('2', profiles)
+      setSelectedProfile(profileToUpdate);
+      setEditRowId(null);
+      setEditedRows(null);
+    } catch (err) {
+      console.error('Error saving profile:', err);
+    }
+  };
+
+  const handleCancel = () => {
+    setEditRowId(null);
+    setEditedRows(null);
+  };
+
+  const handleFieldChange = (e, id) => {
+    const { value, name } = e.target;
+    if (name === 'status' && editRowId === id) {
+      return;
+    }
+    if (name === 'limitedMinutes' && selectedProfile.listWebsites.find(website => website.websiteId._id === id).status !== 'limit') {
+      return;
+    }
+    setEditedRows({
+      ...editedRows,
+      [name]: value
+    });
+  };
+
+  const handleProfileUpdated = async (updatedProfile) => {
+    dispatch(updateProfile(updatedProfile));
+    navigate(0);
+    await fetchProfiles();
+    setSelectedProfile(updatedProfile);
+  };
+
+  const actions = [
+    { func: handleDelete, icon: DeleteIcon, label: 'delete', condition: (id) => id !== editRowId },
+    { func: handleEdit, icon: EditIcon, label: 'edit', condition: (id) => id !== editRowId },
+    { func: handleSave, icon: SaveIcon, label: 'save', condition: (id) => id === editRowId },
+    { func: handleCancel, icon: CancelIcon, label: 'cancel', condition: (id) => id === editRowId },
+  ];
+
+  const generateTableData = (profile) => {
+    if (!profile || !profile.listWebsites || profile.listWebsites.length === 0) {
+      return { headers: [], rows: [] };
+    }
+    const rows = profile.listWebsites
+      .filter(website => website && website.websiteId)
+      .map(website => ({
+        id: website.websiteId ? website.websiteId._id : '',
+        name: (editRowId === website.websiteId._id && editedRows) ? editedRows.name : website.websiteId.name,
+        url: (editRowId === website.websiteId._id && editedRows) ? editedRows.url : website.websiteId.url,
+        status: website.status || '',
+        limitedMinutes: (editRowId === website.websiteId._id && editedRows)
+          ? (editedRows.limitedMinutes === 0 ? '-' : editedRows.limitedMinutes)
+          : (website.limitedMinutes === 0 ? '-' : website.limitedMinutes),
+        Actions: 'Actions',
+      }));
+
+    return {
+      headers: ['name', 'url', 'status', 'limitedMinutes', 'Actions'],
+      rows: rows,
+    };
+  };
+  return (
+    <div className="profile-list-container">
+      <div className="profile-list-select-wrapper">
+        <div className="component-add">
+          <AddProfileComponent />
+        </div>
+        <Select
+          options={profiles.map((profile) => ({ text: profile.profileName, value: profile._id }))}
+          onChange={handleProfileSelect}
+          className="profile-list-select"
+        />
+
+
+
+        {time!==0 &&
+          <div className='timer'>
```

Comment:
timer should be displayed in the profiles table, next to the profile who started the timer